src/main/java/com/exadel/discount/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.exadel.discount.service;
+
+import com.exadel.discount.dto.CategoryDTO;
+import com.exadel.discount.mapper.CategoryMapper;
+import com.exadel.discount.repository.CategoryRepository;
+import com.exadel.discount.service.interfaces.CategoryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public CategoryDTO save(CategoryDTO categoryDTO) {
+        return categoryMapper.getDTO(categoryRepository.save(categoryMapper.parseDTO(categoryDTO)));
+    }
+
+    @Override
+    public CategoryDTO get(UUID id) {
+        return categoryMapper.getDTO(categoryRepository.findById(id).orElse(null));
```

Comment:
Project style is with explicit NotFoundException masseging, as well as loggingn with sjt4j.
Like here:
@Override
    public CategoryDTO get(UUID id) {
        log.debug("Finding Category by ID");

        Category category = categoryRepository
                .findById(id)
                .orElseThrow(() -> new NotFoundException(String.format("Category with id %s not found", id)));

        log.debug("Successfully Category is found by ID");
        return categoryMapper.toCategorynDto(category);
    }

Reply:
+

Reply:
+